File: src/main/java/ac/kr/deu/connect/luck/food_truck/FoodTruckRepository.java
package ac.kr.deu.connect.luck.food_truck;

import ac.kr.deu.connect.luck.user.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface FoodTruckRepository extends JpaRepository<FoodTruck, Long> {
    List<FoodTruck> findByManager(User manager);

    void deleteByManager(User manager);
}

File: src/main/kotlin/ac/kr/deu/connect/luck/foodtruck/repository/FoodTruckRepository.kt
package ac.kr.deu.connect.luck.foodtruck.repository

import ac.kr.deu.connect.luck.foodtruck.entity.FoodTruck
import ac.kr.deu.connect.luck.foodtruck.entity.FoodTruckCategory
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface FoodTruckRepository : JpaRepository<FoodTruck, Long> {

    @Query(
        """
        select foodTruck from FoodTruck foodTruck
        WHERE (:category is null or foodTruck.category = :category) and
              (:managerId is null or foodTruck.managerId = :managerId)
    """
    )
    fun findFoodTrucks(
        category: FoodTruckCategory? = null,
        managerId: Long? = null,
        pageable: Pageable,
    ): Page<FoodTruck>
}
